Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs

```diff
@@ -320,6 +320,28 @@ public void EmbeddedObject_WhenAdditiveExplicit_ShouldThrow()
             Assert.Throws<RealmSchemaValidationException>(() => Realm.GetInstance(conf), $"Embedded object {nameof(EmbeddedLevel3)} is unreachable by any link path from top level objects");
         }
 
+        [Test]
+        public void DeleteRealmWorksIfCalledMultipleTimes()
+        {
+            var config = GetFakeConfig();
+            var openRealm = GetRealm(config);
+            openRealm.Dispose();
+            Assert.That(File.Exists(config.DatabasePath));
+
+            // Race condition:
+            // When trying to delete the Realm it can in some occasions (usually when Sync is involved)
+            // still be in use. To make sure other threads that use the same Realm get scheduled again
+            // and can finish their work before we actually delete the Realm files, we have to wait for
+            // a moment here.
+            // This can be removed as soon as https://github.com/realm/realm-core/issues/4762 is resolved.
```

Comment:
```suggestion
            // TODO: remove this when https://github.com/realm/realm-core/issues/4762 is resolved.
```

Adding a TODO will make it easier to come back to it later.